### PR TITLE
Apply various stubs fixes

### PR DESCRIPTION
**Pull request by mdickinson**
This PR applies various stub fixes; these fixes bring us closer to getting the stubs tests passing on an editable install.

### CODE REVIEW

File: traits/has_traits.pyi [hunk 1]
```diff
@@ -12,7 +12,7 @@ import abc
 from .adaptation.adaptation_error import AdaptationError as AdaptationError
 from .constants import DefaultValue as DefaultValue, TraitKind as TraitKind
 from .ctrait import CTrait as CTrait, __newobj__ as __newobj__
-from .ctraits import CHasTraits as CHasTraits
+from traits.ctraits import CHasTraits as CHasTraits
```
Comment (mdickinson) [inline traits/has_traits.pyi:15]: This was a debugging change, that should be reverted:

```suggestion
from .ctraits import CHasTraits as CHasTraits
```

File: traits/has_traits.pyi [hunk 2]
```diff
@@ -152,8 +152,7 @@ class Vetoable(HasStrictTraits):
 
 VetoableEvent: _Any
 
-class MetaInterface(ABCMetaHasTraits):
-    def __call__(self, adaptee: _Any, default: _Any = ...): ...
+class MetaInterface(ABCMetaHasTraits): ...
```
Comment (mdickinson) [inline traits/has_traits.pyi:155]: The `__call__` signature was causing complaints about disagreement with the parent class. The plan for 7.0 is to get rid of the (long deprecated) special behaviour of `MetaInterface.__call__` anyway, but I'll do that in a separate PR.

File: traits/stubs_tests/examples/Any.py [hunk 3]
```diff
@@ -19,23 +19,23 @@ class Test(HasTraits):
...
-obj.var = "5"
-obj.var = 5
-obj.var = False
-obj.var = 5.5
-obj.var = 5 + 4j
-obj.var = True
-obj.var = [1, 2, "3"]
-obj.var = None
-obj.var = ['1']
+obj.i = {"a": 5, "b": 6}
```
Comment (mdickinson) [inline traits/stubs_tests/examples/Any.py:22]: I don't know whether these `.var` assignments were intentional, but I'd be happy to take a leaf out of NumPy's book and have our initial stubs support _recommended_ use-cases instead of all possible use-cases. In this particular case, most code should be using `HasStrictTraits` rather than `HasTraits`.

File: traits/stubs_tests/examples/completeness.py [hunk 4]
```diff
@@ -21,6 +21,6 @@ class HasName(HasStrictTraits):
 
 def try_assigning_age(x: HasName, new_name: str):
     try:
-        x.age = new_name
+        x.name = new_name
```
Comment (mdickinson) [inline traits/stubs_tests/examples/completeness.py:24]: This was my code; I have _no_ idea what I was thinking.

File: traits/stubs_tests/examples/completeness.py [hunk 5]
```diff
@@ -21,6 +21,6 @@ class HasName(HasStrictTraits):
 
 def try_assigning_age(x: HasName, new_name: str):
```
Comment (mdickinson) [inline traits/stubs_tests/examples/completeness.py:22]:
```suggestion
def try_assigning_name(x: HasName, new_name: str):
```

File: traits/stubs_tests/util.py [hunk 6]
```diff
@@ -108,7 +108,7 @@ def run_mypy(filepath):
         dest_filename = os.path.basename(filepath)
         dest = shutil.copyfile(filepath, os.path.join(tempdir, dest_filename))
         normal_report, error_report, exit_status = mypy_api.run(
-            [dest, '--show-error-code'])
+            [dest, '--show-error-code', '--follow-imports=skip'])
```
Comment (mdickinson) [inline traits/stubs_tests/util.py:111]: The extra argument here was necessary to prevent errors from files _other_ than the one under test. Eventually we should be able to fix the main Traits codebase up enough that we no longer need this, but it may take some time.

File: traits/stubs_tests/util.py [hunk 7]
```diff
@@ -165,5 +165,7 @@ def assertRaisesMypyError(self, filepath):
         for line, error_codes in parsed_mypy_output.items():
             if line not in line_error_map or sorted(
                     line_error_map[line]) != sorted(list(error_codes)):
-                s = "\n{}\n{}".format(str(filepath), normal_report)
+
```
Comment (mdickinson) [inline traits/stubs_tests/util.py:168]:
```suggestion
```

File: traits/stubs_tests/util.py [hunk 8]
```diff
@@ -165,5 +165,7 @@ def assertRaisesMypyError(self, filepath):
...
+                s = "Unexpected error on line {}\n{}\n{}".format(
```
Comment (mdickinson) [inline traits/stubs_tests/util.py:169]: Drive-by improvement to report the line number that caused a test to fail. (Otherwise it's hard to understand what went wrong from the test output.)

File: traits/trait_type.pyi [hunk 9]
```diff
@@ -28,9 +28,6 @@ class _TraitMixin(Generic[_Accepts, _Stores]):
     def __set__(self, object: Any, value: _Accepts) -> None: ...
 
 
-_Trait = _TraitMixin()
```
Comment (mdickinson) [inline traits/trait_type.pyi:31]: I can't tell what this was for, and mypy was complaining about `_Trait` needing a type declaration, so I've deleted it.

File: traits/has_traits.pyi (hunk 2 above)
Comment (mdickinson) [reply traits/has_traits.pyi:155]: > I'll do that in a separate PR.

See #1719

File: traits/stubs_tests/examples/completeness.py (hunk 4 above)
Comment (rahulporuri) [reply traits/stubs_tests/examples/completeness.py:24]: and we will never know. until Enthought succeeds in creating the time machine that is.

**Review by rahulporuri — APPROVED** [review]
LGTM
